lib/ddtrace/tracer.rb

```diff
@@ -160,22 +222,9 @@ def set_tags(tags)
     #   parent2 = tracer.trace('parent2')   # has no parent span
     #   parent2.finish()
     #
+    # This method accepts all the options accepted by \start_span.
     def trace(name, options = {})
-      span = Span.new(self, name, options)
-
-      # set up inheritance
-      parent = @buffer.get()
-      span.set_parent(parent)
-      @buffer.set(span)
-
-      @tags.each { |k, v| span.set_tag(k, v) } unless @tags.empty?
-
-      # sampling
-      if parent.nil?
-        @sampler.sample(span)
-      else
-        span.sampled = span.parent.sampled
-      end
+      span = start_span(name, options)
```

Comment:
OK, got it. Will write a test with the above and comment the implementation to make this obvious. I think I just did not get it in the first place (the fact that `trace()` and `start_span()` had this by design difference about acquiring the context or not).